Sources/Commands/Workspace.swift

```diff
@@ -356,6 +362,43 @@ public class Workspace {
         try saveState()
     }
 
+    /// Ends the edit mode of a dependency which is in edit mode.
+    ///
+    /// - Parameters:
+    ///     - dependency: The dependency to be unedited.
+    ///     - forceRemove: If true, the dependency will be unedited even if has
+    /// unpushed and uncommited changes. Otherwise will throw respective errors.
+    ///
+    /// - throws: WorkspaceOperationError
+    func unedit(dependency: ManagedDependency, forceRemove: Bool) throws {
+        // If the dependency isn't in edit mode, we can't unedit it.
+        guard let basedOn = dependency.basedOn else {
+            throw WorkspaceOperationError.dependencyNotInEditMode
+        }
+        // Form the edit working repo path.
+        let path = editablesPath.appending(dependency.subpath)
+        // Check for uncommited and unpushed changes if force removal is off.
+        if !forceRemove {
+            let workingRepo = try repositoryManager.provider.openCheckout(at: path)
+            guard !workingRepo.hasUncommitedChanges() else {
+                throw WorkspaceOperationError.hasUncommitedChanges(repo: path)
+            }
+            guard try !workingRepo.hasUnpushedCommits() else {
+                throw WorkspaceOperationError.hasUnpushedChanges(repo: path)
+            }
+        }
+        // Remove the editable checkout from disk.
+        try removeFileTree(path)
+        // If this was the last editable dependency, remove the editables directory too.
+        if try localFileSystem.getDirectoryContents(editablesPath).isEmpty {
+            try removeFileTree(editablesPath)
+        }
+        // Restore the dependency state.
+        dependencyMap[dependency.repository] = basedOn
```

Comment:
I'm not sure. I think by default might be ok (I'm not sure what the harm would be), but I also don't think we need to design it here.